api.js

```diff
@@ -972,40 +972,82 @@ const checkMeetingIntersect = function(meetingObj, timeSlotObj) {
-        // }
+        if(hours) {
+            obj.hours = parseInt(String(hours).slice(0, -1));
+        }
+
+        if(mins) {
+            obj.minutes = parseInt(String(mins).slice(0, -1));
+        }
+
+        if(days > 0 || obj.hours >= 4) {
+            obj.hours = 4;
+            obj.minutes = 0;
+        }
 
-        // if(mins) {
-        //     obj.minutes = parseInt(String(mins).slice(0, -1));
-        // }
+        console.log(obj);
 
-        // if(days || obj.hours > 4) {
-        //     obj.hours = 4;
-        //     obj.hours = 0;
-        // }
+        //number of consecutive timeslots needed
+        var slotsToCount = obj.hours * 2 + Math.ceil(obj.minutes / 30); 
+        console.log(slotsToCount);
 
-        // var slotsToCount = obj.hours*2 + obj.minutes/30;
+        let countTimes = 0; //number of suggestions
+        let suggestions = []; //list of timeslot objects
+        let currentSlotCount = 0; //counter for number of consecutive timeslots
 
-        let countTimes = 0;
-        let suggestions = [];
-        //let currentSlotCount = 0;
         //iterate through timetable and find first 5 suggestions
         for(var i = 0; i < userTimes.length; i++) {
             var timeSlot = userTimes[i];
             if(countTimes == 5) {
                 break;
             }
             if((timeSlot.rooms).length != 0) {
-                countTimes++;
-                delete timeSlot.available;
-                suggestions.push(timeSlot);
+                var timeSlotStart = timeSlot.startDateTime;
+                currentSlotCount = 1;
+                var roomSugg = timeSlot.rooms;
+                //console.log(roomSugg);
+                //need to find consecutive timeslots open
+                for(var j = i+1; j < userTimes.length; j++) {
+                    var previousEndTime = new Date(userTimes[j-1].endDateTime).getTime();
+                    var currentStartTime = new Date(userTimes[j].startDateTime).getTime();
+
+                    //if we have a suggestion with rooms that is in succession of the prev time
+                    if(userTimes[j].rooms.length != 0 && previousEndTime == currentStartTime) {
+                        //take the intersection of the rooms
+                        roomSugg = roomSugg.filter(function(n) {
+                            return (userTimes[j].rooms).indexOf(n) > -1;
+                        });
+                        //console.log(roomSugg);
+                        //if the resulting intersect is empty, stop
+                        if(roomSugg.length == 0) {
+                            break;
+                        }
+                        currentSlotCount++;
+                        //if we reached our goal, add it to the list
+                        if(currentSlotCount == slotsToCount) {
+                            countTimes++;
+                            var suggestion = {
+                                startDateTime: timeSlotStart,
```

Comment:
should this be startDateTime or start_datetime?

Reply:
in the API response?  startDateTime works.